packages/next/README.md

```diff
@@ -1879,6 +1881,68 @@ Next.js supports IE11 and all modern browsers out of the box using [`@babel/pres
 
 The [polyfills](https://github.com/zeit/next.js/tree/canary/examples/with-polyfills) example demonstrates the recommended approach to implement polyfills.
 
+## TypeScript
+
+TypeScript is supported out of the box in Next.js. To get started using it create a `tsconfig.json` in your project:
+
+```json
+{
+  "compilerOptions": {
+    "allowJs": true, /* Allow JavaScript files to be type checked. */
+    "jsx": "preserve", /* Preserves jsx outside of Next.js. */
+    "target": "esnext", /* The type checking input. */
+    "module": "esnext", /* Specifies the type of module to type check. */
+    "lib": ["dom", "es2017"], /* List of library files to be included in the type checking. */
+    "noEmit": true, /* Do not emit outputs. Makes sure tsc only does type checking. */
+    "moduleResolution": "node", /* Determine how modules get resolved. */
+    "allowSyntheticDefaultImports": true, /* Allow default imports from modules with no default export. */
```

Comment:
esModuleInterop automatically turns allowSyntheticDefaultImports on, so it is not needed.